medusa/post_processor.py

```diff
@@ -510,51 +510,36 @@ def _add_to_anidb_mylist(self, file_path):
 
     def _find_info(self):
         """
-        For a given file try to find the showid, season, and episode.
+        For a given file try to find the show, season, epsiodes, version and quality.
 
-        :return: A (show, season, episodes, quality, version) tuple
+        :return: A (show, season, episodes, version, quality) tuple
         """
-        show = season = quality = version = None
+        show = season = version = quality = None
         episodes = []
+        name_list = [self.nzb_name, self.file_name, self.rel_path]
 
-        attempt_list = [
-
-            # try to analyze the nzb name
-            lambda: self._analyze_name(self.nzb_name),
-
-            # try to analyze the file name
-            lambda: self._analyze_name(self.file_name),
-
-            # try to analyze the file path
-            lambda: self._analyze_name(self.rel_path)
-        ]
-
-        # Try every possible method to get our info
-        for att_num, attempt in enumerate(attempt_list):
+        for counter, name in enumerate(name_list):
 
-            try:
-                cur_show, cur_season, cur_episodes, cur_quality, cur_version = attempt()
-            except (InvalidNameException, InvalidShowException) as error:
-                logger.log(u'{0}'.format(error), logger.DEBUG)
-                continue
+            cur_show, cur_season, cur_episodes, cur_quality, cur_version = self._analyze_name(name)
 
             if not cur_show:
                 continue
             show = cur_show
 
-            if att_num < (len(attempt_list) - 1):
-                if common.Quality.qualityStrings[cur_quality] == 'Unknown':
-                    continue
-            quality = cur_quality
+            if cur_season is not None:
+                season = cur_season
+
+            if cur_episodes:
```

Comment:
doesn't it need to be `if cur_episodes is not None:` too? otherwise specials won't get PPed ie:
`The.Bachelor.S21E00.Countdown.to.Nick.720p.HDTV.x264-ALTEREGO` or examples from here https://github.com/pymedusa/Medusa/pull/1851